packages/clerk-js/src/core/clerk.ts

```diff
@@ -1003,6 +1010,27 @@ export class Clerk implements ClerkInterface {
     await onAfterSetActive();
   };
 
+  #handlePendingSession = async (session: PendingSessionResource) => {
```

Comment:
@panteliselef @brkalow I think this captures what we have discussed regarding returning earlier on `setActive` to avoid bypassing a lot of the branches of logic by checking the session status 

This method is performing the following:
- Updating session token 
- Performing navigation for UI components vs custom flows
- Updating accessors and emitting listeners

Reply:
I'm auditing this logic now against our unit tests, and it seems we might have to call touch for `pending` still 

On a multi-session application, if you're using an `active` session, but then by the time you switch to another one which is in `pending`, it might have expired and then we need to call `/touch` to handle sign-out